File: controller/comparacion.php
<?php
function compararArchivos($rutaArchivo1, $rutaArchivo2) {
    // Leer el contenido de los archivos
    $contenidoArchivo1 = file($rutaArchivo1);
    $contenidoArchivo2 = file($rutaArchivo2);

    // Comparar los contenidos
    if ($contenidoArchivo1 === $contenidoArchivo2) {
        echo "Los archivos coinciden en un 100%.";
    } else {
        echo "Diferencias encontradas:";
        $lineas1 = count($contenidoArchivo1);
        $lineas2 = count($contenidoArchivo2);
        $numLineas = max($lineas1, $lineas2);

        for ($i = 0; $i < $numLineas; $i++) {
            if ($i < $lineas1 && $i < $lineas2) {
                if ($contenidoArchivo1[$i] !== $contenidoArchivo2[$i]) {
                    echo "Archivo 1: " . $contenidoArchivo1[$i];
                    echo "Archivo 2: " . $contenidoArchivo2[$i];
                    echo "<br>";
                }
            } elseif ($i < $lineas1) {
                echo "Archivo 1: " . $contenidoArchivo1[$i];
                echo "<br>";
            } elseif ($i < $lineas2) {
                echo "Archivo 2: " . $contenidoArchivo2[$i];
                echo "<br>";
            }
        }
    }
}

// Rutas de los archivos a comparar
$rutaArchivo1 = 'archivo1.txt';
$rutaArchivo2 = 'archivo2.txt';

// Llamar a la función para comparar los archivos
compararArchivos($rutaArchivo1, $rutaArchivo2);
?>

File: comparacion.py
def comparar_archivos(archivo1, archivo2):
    # Leer el contenido de los archivos
    with open(archivo1, 'r') as file1:
        contenido1 = file1.readlines()
    with open(archivo2, 'r') as file2:
        contenido2 = file2.readlines()

    # Eliminar los saltos de línea y espacios en blanco al final de cada línea
    contenido1 = [line.rstrip() for line in contenido1]
    contenido2 = [line.rstrip() for line in contenido2]

    # Verificar si los contenidos son iguales
    if contenido1 == contenido2:
        print("Los archivos coinciden en un 100%.")
    else:
        print("Diferencias encontradas:")
        for linea1, linea2 in zip(contenido1, contenido2):
            if linea1 != linea2:
                print(f"Archivo 1: {linea1}")
                print(f"Archivo 2: {linea2}")
                print()

# Rutas de los archivos a comparar
archivo1 = 'archivo1.txt'
archivo2 = 'archivo2.txt'

# Llamar a la función para comparar los archivos
comparar_archivos(archivo1, archivo2)
